src/pgduckdb_ddl.cpp

```diff
@@ -0,0 +1,309 @@
+#include "duckdb.hpp"
+#include <regex>
+
+extern "C" {
+#include "postgres.h"
+#include "access/tableam.h"
+#include "catalog/pg_type.h"
+#include "commands/event_trigger.h"
+#include "fmgr.h"
+#include "catalog/pg_authid_d.h"
+#include "funcapi.h"
+#include "nodes/print.h"
+#include "nodes/makefuncs.h"
+#include "optimizer/optimizer.h"
+#include "tcop/utility.h"
+#include "utils/syscache.h"
+#include "commands/event_trigger.h"
+#include "executor/spi.h"
+#include "miscadmin.h"
+
+#include "pgduckdb/pgduckdb_ruleutils.h"
+}
+
+#include "pgduckdb/pgduckdb_duckdb.hpp"
+#include "pgduckdb/vendor/pg_list.hpp"
+#include <inttypes.h>
+
+/*
+ * Truncates the given table in DuckDB.
+ */
+void
+DuckdbTruncateTable(Oid relation_oid) {
+	auto db = pgduckdb::DuckDBManager::Get().GetDatabase();
+	auto connection = duckdb::make_uniq<duckdb::Connection>(db);
+	auto &context = *connection->context;
+	auto query = std::string("TRUNCATE ") + pgduckdb_relation_name(relation_oid);
+	auto result = context.Query(query, false);
+	if (result->HasError()) {
+		elog(ERROR, "(PGDuckDB/DuckdbTruncateTable) Could not truncate table: %s", result->GetError().c_str());
```

Comment:
Yeah, makes sense. I indeed thought I'd let you figure out a bit how we should handle this in an elegant way, and do it in a crude but working way now. Especially these DDL commands are hopefully not run nearly as often as the regular SELECT queries, so a relatively small leak would probably not be the end of the world.

I think I'll leave it like this for this PR. Especially since your solution isn't perfect yet afaict, because it would still leak the error string. We'd have to copy that over to a memory context, using e.g. `pstrdup` and then manually deallocate (destroy?) the string.